### DISCUSSION TITLE
20241128 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#124] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Roadmap

I've reviewed all feature-request discussions and issues on the main repos (Tracker, Index and Index GUI). I've grouped them and created this diagram:

Impact Mapping: https://miro.com/app/board/uXjVL_SACr0=/

### Torrust website

PR:

- https://github.com/torrust/torrust-website/pull/128

PR review:

- https://github.com/torrust/torrust-website/pull/129

New issues:

- https://github.com/torrust/torrust-website/issues/127
- https://github.com/torrust/torrust-website/issues/126
- https://github.com/torrust/torrust-website/issues/120

### Torrust Tracker

Issues:

- https://github.com/torrust/torrust-tracker/issues/1099

New issues:

- https://github.com/torrust/torrust-tracker/issues/1097
- https://github.com/torrust/torrust-tracker/issues/1099

Commented on issues:

- https://github.com/torrust/torrust-tracker/issues/1069 (proposal for implementation)
- https://github.com/torrust/torrust-tracker/issues/1097#issuecomment-2504306850 (proposal to fix it)

@da2ce7 could you check them?

PR review:

- https://github.com/torrust/torrust-tracker/pull/1095

### Torrust Index

New issues:

- https://github.com/torrust/torrust-index/issues/776
- https://github.com/torrust/torrust-index/issues/775

PR review:

- https://github.com/torrust/torrust-website/pull/125

Related issues:

- https://github.com/torrust/torrust-index-gui/issues/664

Although I opened the issue "[Setup scaffolding for dynamic global settings](https://github.com/torrust/torrust-index-gui/issues/664)" on the Index GUI repo, most of the work needed to enable runtime configuration will be done in the Index. We need to create the corresponding issue on the Index when we start working on that task. That issue was only a draft, and we agreed to introduce user stories in the Index GUI when they have GUI implications.

We will also need to create an issue for "set up scaffolding for user settings," which is the basic work required to implement [user settings](https://github.com/torrust/torrust-index-gui/issues/649).

### Torrust Index GUI

After clustering feature requests, I decided to create two epic issues:

- The most demanded feature requests
- The easy-to-implement features
- Issues we consider essential for `admins` and `users` even if they were not requested by the community

New EPICS:

- https://github.com/torrust/torrust-index-gui/issues/657
- https://github.com/torrust/torrust-index-gui/issues/649

### Torrust Compose

Issues: 

- https://github.com/torrust/torrust-compose/issues/30

**NOTE:** Time spent on Torrust 30h

**Comment by grmbyrn** [comment]
## Torrust Website
* [contributor articles not appearing on contributor page #125](https://github.com/torrust/torrust-website/pull/125) - fix `+layout.server.ts` so data for posts in accessible and relevant blog posts are displayed.
* Replace deprecated Shijiki and shiki-es with Shiki.
* [add component to display contributors #129](https://github.com/torrust/torrust-website/pull/129) - access GitHub API so that a `Contributor.svelte` component can display a grid of contributors to each of the Torrust repos.
* Fix responsiveness on `routes/(pages)` so Table of Contents is above blog post content on smaller screens and on the left on wider screens.
* Work on improving code blocks. Talked with Thomas from Melt UI on why it doesn't have code block component. Found out that Skeleton.dev is deprecating and removing Code Block component in favour of a dedicated integration guide for Shiki.
* Work on [Svelte 5 migration](https://github.com/torrust/torrust-website/issues/106) - Followed [this](https://www.youtube.com/watch?v=_q5FZ3Z9Fz8&t=900s&ab_channel=JoyofCode) but got errors so will inquire on Discord.
* Work on finding styling issues for [this issue](https://github.com/torrust/torrust-website/issues/127) and [this issue](https://github.com/torrust/torrust-website/issues/126).
